CIP-meta-assets (ERC20-like assets)/README.md

```diff
@@ -0,0 +1,426 @@
+
+1 and 2) very much like account based models, wallets supporting this standard will require to know the address of the smart contract (validator)
+
+3) the soultion can co-exsist with the exsiting native tokens
+
+4) the implementaiton is possible without hardfork since Vasil
+
+5) optimized implementations should not take significant computation, especially on transfers.
+
+## Specification
+
+In the specifiaction we'll use the haskell data type `Data`:
+```hs
+data Data
+    = Constr Integer [Data]
+    | Map [( Data, Data )]
+    | List [Data]
+    | I Integer
+    | B ByteString
+```
+
+and we'll use the [plu-ts syntax for structs definition](https://pluts.harmoniclabs.tech/docs/onchain/Values/Structs/definition#pstruct) as an abstraction over the `Data` type.
+
+The core idea of the implementation is to emulate the ERC20 standard; where tokens are entries in a map with addresses (or credentials in our case) as key and integers (the balances) as value. ([see the OpenZeppelin implementation for reference](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/9b3710465583284b8c4c5d2245749246bb2e0094/contracts/token/ERC20/ERC20.sol#L16));
+
+Unlike the ERC20 standard; this CIP:
+
+- allows for multiple entries with the same key (same credentials can be used for multiple accounts)
+- DOESN'T include an equivalent of the `transferFrom` method; if needed it can be added by a specific implementation but it won't be considered part of the standard.
+
+> **NOTE**
+>
+> the UTxO model allows for multiple transfers in one transaction
+>
+> this would allow for a more powerful model than the account based equivalent but implies higher execution costs
+>
+> whith the goal of keeping the standard interoperable and easy to understand and implement
+> in this first implementation we restricts transfers from a single account to a single account
+>
+> if necessary; this restriction might be dropped in a future version of the CIP 
+
+
+
+The implementation requires
+
+- a parametrized minting policy to validate the creation of new accounts (in this CIP also referred as `accountFactory`)
+- a spending validator to manage the accounts ((in this CIP also referred as `accountManager`))
+
+### standard data types used
+
+We'll need to use some data types as defined in the script context;
+
+```ts
+const PTxId = pstruct({
+    PTxId: { txId: bs }
+});
+
+const PTxOutRef = pstruct({
+    PTxOutRef: {
+        id: PTxId.type,
+        index: int
+    }
+});
+
+const PCredential = pstruct({
+    PPubKeyCredential: { pkh: bs },
+    PScriptCredential: { valHash: bs },
+});
+
+const PCurrencySymbol = palias( bs );
+```
+which are equivalent to the data:
+```hs
+-- PTxId
+Constr 0 [ B txId ]
+
+--- PTxOutRef
+Constr 0 [ PTxId, I index ]
+
+-- PPubKeyCredential
+Constr 0 [ B pkh ]
+
+-- PScriptCredential
+Constr 1 [ B valHash ]
+
+-- PCurrencySymbol
+B currencySym
+```
+
+## `accountFactory` (minting policy)
```

Comment:
```suggestion
### `accountFactory` (minting policy)
```

Reply:
@michele-nuzzi this is a straightforward header problem, and mandatory to fix, so let's commit it & move on to more significant issues 🤓